test/ExportQueryExecutionTreesTest.cpp

```diff
@@ -1637,3 +1637,104 @@ TEST(ExportQueryExecutionTrees, convertGeneratorForChunkedTransfer) {
               AllOf(HasSubstr("!!!!>># An error has occurred"),
                     HasSubstr("A very strange")));
 }
+
+TEST(ExportQueryExecutionTrees, idToLiteralOrIriFunctionality) {
+  std::string kg =
+      "<s> <p> \"something\" . <s> <p> 1 . <s> <p> "
+      "\"some\"^^<http://www.w3.org/2001/XMLSchema#string> . <s> <p> "
+      "\"dadudeldu\"^^<http://www.dadudeldu.com/NoSuchDatatype> .";
+  auto qec = ad_utility::testing::getQec(kg);
+  auto getId = ad_utility::testing::makeGetId(qec->getIndex());
+  using enum Datatype;
+
+  // Case Literal With no Datatype
+  {
+    Id id = getId("\"something\"");
+    auto resultLiteral = ExportQueryExecutionTrees::idToLiteralOrIri(
+        qec->getIndex(), id, LocalVocab{});
+    EXPECT_EQ(resultLiteral.value().toStringRepresentation(), "\"something\"");
+    // Case onlyReturnLiterals
+    resultLiteral = ExportQueryExecutionTrees::idToLiteralOrIri<true>(
```

Comment:
There is a lot of code duplication here.
As a first step you can factor out the calls to `idToLiteralOrIri` (write a lambda that does this for you, such that the calls get shorter.

Then you can look up the gtest matchers `::testing::Optional` and `::testing::ResultOf` to get this shorter (the googletest documentation, or the rest of the codebase can give you hints, otherwise contact me.